src/disk_io_thread.cpp

```diff
@@ -941,12 +946,19 @@ namespace libtorrent
 			--pe->piece_refcount;
```

Comment:
the call to try_flush_hashed() may also write to disk, no? which I believe currently would not participate in the condition variable